library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -14,37 +14,54 @@
 
 package kwasm.format.text
 
+import kwasm.ast.ValueType
 import kwasm.format.ParseContext
+import kwasm.format.ParseException
 
-sealed class Type(
+sealed class Type<T>(
     protected val sequence: CharSequence,
     protected val context: ParseContext? = null
 ) {
+    val value: T by lazy {
+        parseValue()
+    }
 
-    protected abstract fun parseValue(): Unit
+    protected abstract fun parseValue(): T
 
     class ValueType(
         sequence: CharSequence,
         context: ParseContext? = null
-    ) : Type(sequence, context) {
-        override fun parseValue() {
-            TODO("not implemented")
+    ) : Type<kwasm.ast.ValueType>(sequence, context) {
```

Comment:
KDoc would be good to add here, and maybe a short one for the `Type` class itself.

Reply:
Done